scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1052,6 +1052,27 @@ public boolean accept(Principal connectionPrincipal) {
 		return startTerminateConnectionsForPrincipal(handler);
 	}
 
+	/**
+	 * Start to terminate connections applying the provided handler to the
+	 * principals of all connections.
+	 * 
+	 * Note: if {@link SessionCache} is used, it's not possible to remove a
+	 * cache entry, if no related connection is in the connection store. All
+	 * available connections will be removed from that session cache as well.
+	 * 
+	 * @param principalHandler handler to be called within the serial execution
+	 *            of the related connection. If {@code true} is returned, the
+	 *            related connection is terminated and the session is removed
+	 *            from the session cache.
+	 * @return future to cancel or wait for completion
+	 * @see #startTerminateConnectionsForPrincipal(org.eclipse.californium.elements.util.LeastRecentlyUsedCache.Predicate,
+	 *      boolean)
+	 */
+	public Future<Void> startTerminateConnectionsForPrincipal(
+			LeastRecentlyUsedCache.Predicate<Principal> principalHandler) {
```

Comment:
Just some thought about this.
In our case we have a `SessionCache` (RedisSessionCache).
And for some reason we want to remove connection+session for a given DTLS Identity. (e.g. compromised credentials)

As you said in you comment : 
> Note: if **SessionCache** is used, it's not possible to remove a
 cache entry, if no related connection is in the connection store. All
 available connections will be removed from that session cache as well.

My current understanding is in common use case, if you are using SessionCache you will face the situation where connection is not in connection store. Meaning that if you want to remove session you should do it separately.

Does it means that maybe the right default value for this method should be `removeFromCache = false` ?
(Of course we should not change it for 2.x but maybe for 3.x ?)

Reply:
Hm, for me these are details for #1345